irohad/pending_txs_storage/impl/pending_txs_storage_impl.hpp

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_PENDING_TXS_STORAGE_IMPL_HPP
+#define IROHA_PENDING_TXS_STORAGE_IMPL_HPP
+
+#include <set>
+#include <shared_mutex>
+#include <unordered_map>
+#include <unordered_set>
+
+#include <rxcpp/rx.hpp>
+#include "pending_txs_storage/pending_txs_storage.hpp"
+#include "interfaces/iroha_internal/transaction_batch.hpp"
+
+namespace iroha {
+
+  class MstState;
+
+  class PendingTransactionStorageImpl : public PendingTransactionStorage {
+   public:
+    using AccountIdType = shared_model::interface::types::AccountIdType;
+    using HashType = shared_model::interface::types::HashType;
+    using SharedTxsCollectionType =
+        shared_model::interface::types::SharedTxsCollectionType;
+    using TransactionBatch = shared_model::interface::TransactionBatch;
+    using SharedState = std::shared_ptr<MstState>;
+    using SharedBatch = std::shared_ptr<TransactionBatch>;
+    using StateObservable = rxcpp::observable<SharedState>;
+    using BatchObservable = rxcpp::observable<SharedBatch>;
+
+    PendingTransactionStorageImpl(StateObservable updatedBatches,
+                                  BatchObservable preparedBatch,
+                                  BatchObservable expiredBatch);
+
+    ~PendingTransactionStorageImpl() override;
+
+    SharedTxsCollectionType getPendingTransactions(
+        const AccountIdType &accountId) const override;
+
+   private:
+    void updatedBatchesHandler(const SharedState &updatedBatches);
+    void preparedBatchHandler(const SharedBatch &preparedBatch);
+    void expiredBatchHandler(const SharedBatch &expiredBatch);
+
+    void removeBatch(const SharedBatch &batch);
+
+    std::set<AccountIdType> batchCreators(const TransactionBatch &batch) const;
+
+    /**
+     * Subscriptions on MST events
```

Comment:
This comment looks related only to one subscription. I think better to use `///` instead.
Or use it with this manner:
```
// ---------| Subscriptions on MST events |---------
```
Of course, better to wrap it by 80 symbols for clean-up. Also, you may request me for the python generator of those comments.

Reply:
Will leave it as is. Subscriptions + comment are separated from the rest of code with empty lines. In my view, it is pretty clear that comment relates to all subscription objects. Even more, comment is written in plural form. Thus, everything here is ok.

Reply:
I missed a bit with the comment, I imagine using `//` as in the example.